### PR TITLE
Show user OpenID URIs in their profile

### PR DESCRIPTION
**Pull request by strk**
So you can tell from a quick look if the users are really who you think
they are (if you know their openid)

### CODE REVIEW

**Comment by lunny** [comment]
maybe you could send a screenshot?

**Comment by strk** [comment]
@lunny:

show-openid

**Comment by ptman** [comment]
looks good

**Comment by lunny** [comment]
text align ...

**Comment by strk** [comment]
What about text align @lunny ?
Looks aligned here

**Review by appleboy — APPROVED** [review]

**Comment by lunny** [comment]
@strk 
s
h
j
0

**Comment by strk** [comment]
@lunny I don't understand your message

**Comment by Fastidious** [comment]
Why is it bold? What happens when the URI is `https://this.is.my.openid.provider.of.choice.which.i.use.here/bananas`? If the OpenID is to authenticate/login, why is there a need to show it? What is the benefit of it?

**Comment by strk** [comment]
Your OpenID basically shows who you are, so it's part of your identity.
With OpenID that identity is "verified" (by Gitea, in this case).
So you're saying the world you are "my.banan.as", and by showing it
Gitea is telling the visitor that it verified you are.

Good question about bold, doesn't need be. Changed with 9c7ccc2

What happens when URL is long ? I guess the display is bad,
want to test it ? I can setup a staging install with it.

**Comment by strk** [comment]
Made a test with long OpenID (and no-more-bold). Here's what you get:

long-openid

Maybe there's some client-side trick to adapt to screen in javascript ? @geek1011

**Comment by Fastidious** [comment]
Also, http://fontawesome.io/icon/openid/. The icon being used now is a blur of rubbish. 😄

**Comment by strk** [comment]
7ebf154 uses font-awesome, and here's a screenshot from another machine (with non-magnified browser content):

fa-openid

**Comment by strk** [comment]
And those links become clikable with f5b8cb6, and here's another screenshot:

clickable-openid

**Comment by strk** [comment]
NOTE: from this other machine (Firefox) the overflow *never* happens (the first screenshot with overflow was from another browser, dunno now which one it was)

**Comment by Fastidious** [comment]
@strk you wrote:
> Your OpenID basically shows who you are [...]

I though OpenID was about authentication, not identification. Am I wrong?

**Comment by lunny** [comment]
@strk maybe you should give an option to enable to show OpenID in the profile and default is disabled. Security is always the first class Gitea should consider.

**Comment by strk** [comment]
@Fastidious I think OpenID can serve both. By all means authentication is authentication of an identity (your URL). @lunny good point about allowing a flag to make visible/hidden, will work on it (maybe per-openid).
Speaking of which, I went looking how that's done for email and found a bug (#1323)

**Comment by strk** [comment]
Ready, there's now a per-OpenID show/hide toggle, defaulting to hide.

**Comment by strk** [comment]
show-hide-openid

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -357,6 +357,8 @@ key_name = Key Name
 key_content = Content
 add_key_success = Your new SSH key '%s' has been added successfully!
 delete_key = Delete
+show_key = Show
```
Comment (pgaskin) [inline options/locale/locale_en-US.ini]: It is kind of unclear what this does, maybe make it "Show on profile"

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -357,6 +357,8 @@ key_name = Key Name
...
+hide_key = Hide
```
Comment (pgaskin) [inline options/locale/locale_en-US.ini]: It is kind of unclear what this does, maybe make it "Hide on profile"

**Review by pgaskin — APPROVED** [review]
LGTM except for my comments

**Comment by pgaskin** [comment]
Make LGTM work

**Comment by strk** [comment]
Like this, @geek1011 ?

showhide-long

File: options/locale/locale_en-US.ini (hunk 2 above)
Comment (strk) [reply options/locale/locale_en-US.ini]: See how you like a117987

**Comment by strk** [comment]
I've also added unit test now.
Oh, and @Fastidious pointed out he thinks the text should be `Hide in profile` rather than `Hide from profile`, what do others think ?

**Comment by lunny** [comment]
I have no idea about `Hide in profile` or `Hide from profile`, after that, I will give it LGTM.

**Comment by pgaskin** [comment]
@lunny `Hide in profile` means you are putting it somewhere in the profile, and `hide from profile` means that you are not including it in the profile. I speak English as my first language.

**Comment by bkcsoft** [comment]
Conflicting migrations. Otherwise LGTM

**Comment by bkcsoft** [comment]
poof?

**Pull request merged**